uaclient/api/u/pro/detect/should_auto_attach/v1.py

```diff
@@ -0,0 +1,41 @@
+from uaclient import exceptions
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.apt import is_installed
+from uaclient.clouds.identity import cloud_instance_factory
+from uaclient.config import UAConfig
+from uaclient.data_types import BoolDataValue, DataObject, Field
+
+
+class ShouldAutoAttachResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("should_auto_attach", BoolDataValue),
+    ]
+
+    def __init__(self, should_auto_attach: bool):
+        self.should_auto_attach = should_auto_attach
+
+
+def should_auto_attach_on_machine(cfg: UAConfig):
```

Comment:
I expected this to match the api endpoint name: `should_auto_attach`. Do we need the "_on_machine"?

Reply:
Yeah, this is just so we avoid naming the function and the Result object attribute the same, but I guess it is better to change the attribute name instead

Reply:
@lucasmoura I wouldn't mind if the function and the attribute have the same name :upside_down_face:

Reply:
Yeah, good point. I think this is not a concern at all. Changing it